modules/core/src/main/java/org/apache/ignite/internal/processors/rest/handlers/cache/GridCacheCommandHandler.java

```diff
@@ -563,18 +570,14 @@ private static Object appendOrPrepend(Object origVal, Object appendVal,
                     // HashSet wrapping for correct serialization
                     Set<Object> keys = map == null ? null : new HashSet<>(map.keySet());
 
-                    fut = executeCommand(req.destinationId(), req.clientId(), cacheName, cacheFlags, key,
+                    fut = executeCommand(req.destinationId(), req.clientId(), req0.cacheName(), cacheFlags, key,
                         new RemoveAllCommand(keys));
 
                     break;
                 }
 
                 case CACHE_CLEAR: {
-                    Map<Object, Object> map = req0.values();
-
-                    // HashSet wrapping for correct serialization
-                    Set<Object> cacheNames = map == null ?
-                        new HashSet<Object>(ctx.cache().publicCacheNames()) : new HashSet<>(map.keySet());
+                    Set<Object> cacheNames = getKeys(req0);
```

Comment:
It was values and know it's keys? Why?

Reply:
Do you mean that now I check keys? GetKeys() function does exactly the same as code that I've removed (except getting of all public caches when values() are null).

Reply:
Do you mean that now I check keys? GetKeys() function does exactly the same as code that I've removed (except getting of all public caches when values() are null)